api/node/package.json

```diff
@@ -23,7 +23,7 @@
   },
   "scripts": {
     "artifacts": "napi artifacts",
-    "compile": "esbuild index.ts --bundle --external:*.node --format=cjs --platform=node --outfile=index.js",
+    "compile": "esbuild index.ts --bundle --external:*.node --format=cjs --platform=node --outfile=index.js && tsc --declaration --emitDeclarationOnly",
```

Comment:
Calling tsc here means two things now:

1) the typescript compiler is now an install dependency but it isn’t explicitly listed as such.
2) types packages are missing in the install dependencies, which breaks the nightly packages: https://github.com/slint-ui/slint/actions/runs/6852766341/job/18632057437

either the dependencies need to be fixed or we should compile once (including types) and ship that, instead of compiling at install time (we can still compile the native code).

I think the second option is the idiomatic path, but it’s more work.